baiji/util/tempfile.py

```diff
@@ -7,10 +7,13 @@
 '''
 from __future__ import absolute_import
 import os
+from baiji.util.shutillib import mkdir_p
 
 def _override_tmp(fn, args, kwargs):
     if 'dir' not in kwargs:
         kwargs['dir'] = os.environ.get('BAIJI_TMP', None)
+        if kwargs['dir'] is not None:
+            mkdir_p(kwargs['dir'])
```

Comment:
That's why I'm using `os.environ.get`, not `os.environ['BAIJI_TMP']`. The second argument to `get` is a default value for when the key is not set (works on all dicts, actually, not just `environ`). If the variable is not set, then `kwargs['dir']` will be set to None, which is the default and causes mkstmp to use the system temp dir. When were using the system temp dir, mkstmp is already making sure it exists, so we need do nothing.